libgit2-glib: new spack recipe for gnome package.

KineticTheory opened the pull request:
**New package:**  `libgit2-glib` is a `glib` wrapper library around the `libgit2` git access library. `libgit2` only implements the core plumbing functions, not really the higher-level porcelain stuff.

<!--  
Remember that `spackbot` can help with your PR in multiple ways:
- `@spackbot help` shows all the commands that are currently available
- `@spackbot fix style` tries to push a commit to fix style issues in this PR
- `@spackbot re-run pipeline` runs the pipelines again, if you have write access to the repository 
-->

rbberger commented on repos/spack_repo/builtin/packages/libgit2_glib/package.py:
```diff
@@ -0,0 +1,45 @@
+
+    maintainers("KineticTheory")
+
+    license("GNU LGPL-2.1")
```
This should be:
```suggestion
    license("LGPL-2.1-or-later")
```

KineticTheory replied:
Fixed! TY.

KineticTheory commented:
@rbberger I implemented your suggested change last week.  I think this is ready for a 2nd look and possible merge.  TIA.

rbberger commented on repos/spack_repo/builtin/packages/libgit2_glib/package.py:
```diff
@@ -0,0 +1,45 @@
+
+class MesonBuilder(meson.MesonBuilder):
+    def meson_args(self):
+        args = ["-Dssh=false"]
```
It might be worth expanding this boolean from `self.spec.satisfies("^libgit2+ssh)`. Similarly, other dependencies are only needed for certain [options](https://gitlab.gnome.org/GNOME/libgit2-glib/-/blob/master/meson_options.txt?ref_type=heads), which could be mapped as variants.

KineticTheory replied:
@rbberger I've added variants to support some of the other meson build options including logic to enable ssh support.  I did not add a variant `+docs` because this build mode requires additional packages that do not yet exist in the spack ecosystem.

KineticTheory committed:
libgit2-glib: new spack recipe for gnome package

+ New package: libgit2-glib is a glib wrapper library around the libgit2
  git access library. libgit2 only implements the core plumbing functions,
  not really the higher-level porcelain stuff.
+ Add support for variants `~vapi`, `~python`, and `^libgit2+ssh`.

KineticTheory committed:
Style fix.
